### DISCUSSION TITLE
While editing Readme, Back button unconditionally discards changes

### DISCUSSION THREAD

**Discussion by Doomster14** [community/community#118994] [Repositories]
### Select Topic Area

Bug

### Body

The following scenario leads to a loss of input:

- Open a repo page, like https://github.com/Doomster14/CommonSenseExpansion.
- Press "Edit file" button to edit `README.md`.
- Make some edits.
- Press the Back button in browser.
- On the warning "This page is asking you to confirm that you want to leave — information you’ve entered may not be saved.", answer "Stay on page".
- On the question "You have unsaved changes. Do you want to discard them?", choose "Cancel".
- Oops, you're back on the repo page, and the edits are lost.

Browser: Firefox 115.9.1esr (64-bit)

For me, this works normally if I try to leave the editor page by clicking links or opening bookmarks. I.e., I get a warning, I choose to stay, and I'm allowed to stay. Only the Back button is bugged, afaik.

**Comment by yc3yu** [comment]
The same thing happened to me on Chrome, and I lost all my edits to the README file that I made over the past couple of days (which is a lot since I'm rewriting my repo's README from scratch). Is there any way to recover my work?

**Comment by samuelchen** [comment]
The same!
